### PR TITLE
Migrate update main image background job to sidekiq

### PR DESCRIPTION
**Pull request by codekatas-magmalabs**
## What type of PR is this? (check all applicable)

- [x] Feature
- [x] Optimization

## Related Tickets & Documents
https://github.com/thepracticaldev/dev.to/issues/5305

## Added to documentation?

- [x ] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

flanders

### CODE REVIEW

File: app/workers/articles/update_main_image_background_hex_worker.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module Articles
+  class UpdateMainImageBackgroundHexWorker
+    include Sidekiq::Worker
+
+    sidekiq_options queue: :high_priority, retry: 10
```
Comment (rhymes) [inline app/workers/articles/update_main_image_background_hex_worker.rb]: @mstruve what do you think about this priority?

**Comment by rhymes** [comment]
Also, don't forget to sign the CLA up here

File: app/workers/articles/update_main_image_background_hex_worker.rb (hunk 1 above)
Comment (mstruve) [reply app/workers/articles/update_main_image_background_hex_worker.rb]: I do think after a user saves their article they will want to see it, with proper colors, right away so this seems like the right queue. We can always adjust later.

**Comment by atsmith813** [comment]
Hey @codekatas-magmalabs, thanks for the contribution! Would you like to update the specs as Rhymes mentioned? I know life has funny ways of getting busy. I'm happy to update it or assist as well if that's easier, just let me know. Thanks!

File: app/models/article.rb [hunk 2]
```diff
@@ -429,7 +429,7 @@ def evaluate_markdown
   def update_main_image_background_hex
```
Comment (mstruve) [inline app/models/article.rb]: This is also being triggered in an `after_save` callback which means the record will not be in the database yet if it is new. Can you change this so it's fired in an `after_commit` callback? THANK YOU!

**Comment by atsmith813** [comment]
@codekatas-magmalabs I went ahead and pushed up the requested changes. We're in the home stretch of finishing off this epic (#5305). Thank you so much for getting this going and please let me know if you have any questions!

@mstruve @rhymes I've made the requested changed!

**Pull request merged**